pkg/transport/manager.go

```diff
@@ -297,7 +293,7 @@ func (tm *Manager) SaveTransport(ctx context.Context, remote cipher.PubKey, tpTy
 				if closeErr := mTp.Close(); closeErr != nil {
 					tm.Logger.WithError(err).Warn("Closing mTp returns non-nil error.")
 				}
-				delete(tm.tps, mTp.Entry.ID)
+				tm.DeleteTransport(mTp.Entry.ID)
```

Comment:
I'm not sure that it's safe to call `DeleteTransprot` here. Previously we just removed tp from the internal map here, while `DeleteTransport` also has responsibility to remove it from discovery which is not intended here I guess. Also `DeleteTransport` tries to close tp which is already closed at the point of this call. I guess it would be better to add a separate func starting with lowercase character like `deleteTransport` which just removes tp from the map under lock. Also this func should be called below in a goroutine on line 355